src/core/reference/src/op/concat.cpp

```diff
@@ -44,5 +44,33 @@ void concat(const std::vector<const char*>& args,
         }
     }
 }
+
+void concat(const std::vector<const std::string*>& args,
+            std::string* out,
+            const std::vector<Shape>& in_shapes,
+            const Shape& out_shape,
+            int64_t concatenation_axis,
+            size_t) {
+    size_t steps = 1;
+    for (int i = 0; i < concatenation_axis; ++i) {
+        steps *= out_shape[i];
+    }
```

Comment:
steps is just shape size
```suggestion
    const size_t steps = concatenation_axis < 0 ? 1 : shape_size(out_shape.begin(), out_shape.begin() + concatenation_axis);
```
`concatenation_axis < 0` could be not required as normalized value is passed